src/main/java/it/pagopa/pm/gateway/service/CcResumeService.java

```diff
@@ -0,0 +1,193 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+
+import static it.pagopa.pm.gateway.constant.Messages.GENERIC_ERROR_MSG;
+import static it.pagopa.pm.gateway.constant.Messages.PATCH_CLOSE_PAYMENT_ERROR;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_AUTHORIZED;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_CHALLENGE;
+import static it.pagopa.pm.gateway.dto.enums.PaymentRequestStatusEnum.*;
+import static it.pagopa.pm.gateway.dto.enums.TransactionStatusEnum.TX_AUTHORIZED_BY_PGS;
+import static it.pagopa.pm.gateway.dto.enums.TransactionStatusEnum.TX_REFUSED;
+
+@Service
+@Slf4j
+public class CcResumeService {
+
+    @Value("${vpos.requestUrl}")
+    private String vposUrl;
+
+    @Autowired
+    private PaymentRequestRepository paymentRequestRepository;
+
+    @Autowired
+    private RestapiCdClientImpl restapiCdClient;
+
+    @Autowired
+    private VPosRequestUtils vPosRequestUtils;
+
+    @Autowired
+    private VPosResponseUtils vPosResponseUtils;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public void startResume(CreditCardResumeRequest request, String requestId) {
+        PaymentRequestEntity entity = paymentRequestRepository.findByGuid(requestId);
+
+        if (Objects.isNull(entity)) {
+            log.error("No CreditCard request entity has been found for requestId: " + requestId);
+            return;
+        }
+
+        if (Objects.nonNull(entity.getAuthorizationOutcome())) {
+            log.warn(String.format("requestId %s already processed", requestId));
+            entity.setErrorMessage("requestId already processed");
+            return;
+        }
+        processResume(request, entity, requestId);
+    }
+
+    private void processResume(CreditCardResumeRequest request, PaymentRequestEntity entity, String requestId) {
+        String methodCompleted = request.getMethodCompleted();
+        String responseType = entity.getResponseType();
+        String correlationId = entity.getCorrelationId();
+        try {
+            StepZeroRequest stepZeroRequest = objectMapper.readValue(entity.getJsonRequest(), StepZeroRequest.class);
+            stepZeroRequest.setIsFirstPayment(false);
+            MethodCompletedEnum methodCompletedEnum = StringUtils.isBlank(methodCompleted) ? null : MethodCompletedEnum.valueOf(methodCompleted);
+            if (Objects.nonNull(responseType) && responseType.equalsIgnoreCase(ThreeDS2ResponseTypeEnum.METHOD.name())) {
+                Map<String, String> params = vPosRequestUtils.buildStepOneRequestParams(methodCompletedEnum, stepZeroRequest, correlationId);
```

Comment:
Here `methodCompletedEnum` can be null:
* Forwarded to `buildStepOneRequestParams`
* Forwarded to `buildStepOneRequest`
* `name` method gets called, triggering a `NullPointerException`

Reply:
I removed control and the exception is catched